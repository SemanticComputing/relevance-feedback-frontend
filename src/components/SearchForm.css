.disabledButton {
  border: 1px solid #0066cc;
  background-color: #774444 !important;
  color: #888888 !important;
}

.activeButton {
  border: 1px solid #0066cc;
}
