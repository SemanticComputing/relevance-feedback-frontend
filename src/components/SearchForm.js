import React from 'react';
import TextInput from '../components/TextInput';
import { object, func } from 'prop-types';
import { Form, InputGroup, Label, Button } from 'reactstrap';

const SearchForm = ({ updateQuery, doSearch, search }) => {

  const handleSubmit = (event) => {
    event.preventDefault();
    return doSearch(search);
  };

  return (
    <Form onSubmit={handleSubmit}>
      <InputGroup>
        <Label hidden for='search'>Haku</Label>
        <TextInput id='search' value={search.query} placeholder='Haku' onChange={updateQuery} />
        <Button disabled={search.disabled} type="submit">Hae</Button>
      </InputGroup>
    </Form>
  );
};

SearchForm.propTypes = {
  search: object,
  updateQuery: func,
  doSearch: func,
};

export default SearchForm;
