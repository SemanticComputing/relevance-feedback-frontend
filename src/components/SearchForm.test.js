import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import SearchForm from './SearchForm';

const search = { query: 'query' };

it('renders without crashing', () => {
  const doSearch = jest.fn();
  const updateQuery = jest.fn();

  const div = document.createElement('div');
  ReactDOM.render(<SearchForm doSearch={doSearch} updateQuery={updateQuery} search={search} />, div);
});

it('displays the query', () => {
  const doSearch = jest.fn();
  const updateQuery = jest.fn();

  const tree = renderer
    .create(<SearchForm doSearch={doSearch} updateQuery={updateQuery} search={search} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('calls doSearch on form submit', () => {
  const doSearch = jest.fn();
  const updateQuery = jest.fn();

  const form = mount(<SearchForm doSearch={doSearch} updateQuery={updateQuery} search={search} />);

  form.find('form').simulate('submit');

  expect(doSearch.mock.calls.length).toBe(1);
  expect(doSearch.mock.calls[0][0]).toEqual(search);
});

it('calls updateQuery when on input change', () => {
  const updateQuery = jest.fn();
  const doSearch = jest.fn();

  const form = mount(<SearchForm doSearch={doSearch} updateQuery={updateQuery} search={search} />);

  form.find('input').simulate('change', { target: { value: 'changed' } });

  expect(updateQuery.mock.calls.length).toBe(1);
  expect(updateQuery.mock.calls[0][0]).toEqual('changed');
  expect(doSearch.mock.calls.length).toBe(0);
});
