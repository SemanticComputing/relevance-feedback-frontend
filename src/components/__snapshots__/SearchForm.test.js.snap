// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays the query 1`] = `
<form
  className=""
  onSubmit={[Function]}
>
  <div
    className="input-group"
  >
    <label
      className="sr-only  form-control-label"
      htmlFor="search"
    >
      Haku
    </label>
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="Haku"
      type="text"
      value="query"
    />
    <button
      className="btn btn-secondary"
      disabled={false}
      onClick={[Function]}
      type="submit"
    >
      Hae
    </button>
  </div>
</form>
`;
